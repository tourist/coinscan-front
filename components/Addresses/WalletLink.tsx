import Link from '../Link';
import Box from '@mui/material/Box';

import Hash from './Hash';
import ChainScannerLink from '../Addresses/ChainScannerLink';
import CopyToClipboard from '../CopyToClipboard';
import settings from '../../settings.json';

type WalletLinkProps = {
  walletToLink: string;
  scannerLink?: boolean;
  currentWallet?: string;
  short?: boolean;
};

const WalletLink = ({
  currentWallet,
  walletToLink,
  scannerLink,
  short = false,
}: WalletLinkProps) => {
  const link = `${settings.scannerAddressLink}${walletToLink}`;
  return (
    <Box
      sx={{
        display: 'flex',
        flexDirection: 'row',
        alignItems: 'center',
        a: { display: 'flex' },
      }}
    >
      {currentWallet === walletToLink ? (
        <Hash short={short} text={walletToLink} />
      ) : (
        <Link
          title={walletToLink}
          href={{
            pathname: '/wallet/[address]',
            query: { address: walletToLink },
          }}
        >
          <Hash
            text={
              settings.addresses[walletToLink]
                ? settings.addresses[walletToLink]
                : walletToLink
            }
            title={walletToLink}
            short={short}
          />
        </Link>
      )}
      <CopyToClipboard text={walletToLink} />
      {scannerLink ? (
        <>
          <Box sx={{ ml: 1 }} />
          <ChainScannerLink title={link} link={link} />
        </>
      ) : null}
    </Box>
  );
};

export default WalletLink;
