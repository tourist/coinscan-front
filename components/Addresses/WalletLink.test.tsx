import { screen, render } from '@testing-library/react';
import snapshotDiff from 'snapshot-diff';
import WalletLink from './WalletLink';

test('render wallet link without link', async () => {
  const { container } = render(
    <WalletLink
      currentWallet="0x320a50f32fb9e20fe113573unknown"
      walletToLink="0x320a50f32fb9e20fe113573unknown"
    />
  );
  screen.getByText('0x320a50f32fb9e20fe113573unknown');
  expect(container).toMatchSnapshot();
});

test('render wallet link with link when currentWallet passed', async () => {
  const { container } = render(
    <WalletLink
      currentWallet="0x320a50f32fb9e20fe113573current"
      walletToLink="0x320a50f32fb9e20fe113573other"
    />
  );
  screen.getByText('0x320a50f32fb9e20fe113573other');
  expect(
    screen.getAllByTitle('0x320a50f32fb9e20fe113573other')[0]
  ).toHaveAttribute('href', '/wallet/0x320a50f32fb9e20fe113573other');
  expect(container).toMatchSnapshot();
});

test('render wallet link without link when currentWallet passed', async () => {
  const { container } = render(
    <WalletLink
      currentWallet="0x320a50f32fb9e20fe113573other"
      walletToLink="0x320a50f32fb9e20fe113573other"
    />
  );
  screen.getByText('0x320a50f32fb9e20fe113573other');
  expect(container).toMatchSnapshot();
});

test('render wallet with external link to scanner', async () => {
  const { container } = render(
    <WalletLink walletToLink="0x320a50f32fb9e20fe113573" scannerLink />
  );
  screen.getByText('0x320a50f32fb9e20fe113573');
  expect(
    screen.getAllByTitle(
      'https://polygonscan.com/address/0x320a50f32fb9e20fe113573'
    )[0]
  ).toHaveAttribute(
    'href',
    'https://polygonscan.com/address/0x320a50f32fb9e20fe113573'
  );
  expect(container).toMatchSnapshot();
});

test('render wallet with short prop set', async () => {
  const { asFragment, rerender } = render(
    <WalletLink walletToLink="0x320a50f32fb9e20fe113573" />
  );
  const firstRender = asFragment();
  rerender(<WalletLink walletToLink="0x320a50f32fb9e20fe113573" short />);
  screen.getByText('0x320a50f32fb9e20fe113573');
  expect(snapshotDiff(firstRender, asFragment())).toMatchSnapshot(
    'rerender with short prop'
  );
});
