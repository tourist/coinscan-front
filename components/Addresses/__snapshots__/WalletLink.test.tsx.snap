// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render wallet link with link when currentWallet passed 1`] = `
<div>
  <div
    class="MuiBox-root css-1gvmo7"
  >
    <a
      class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1nrvlnd-MuiTypography-root-MuiLink-root"
      href="/wallet/0x320a50f32fb9e20fe113573other"
      title="0x320a50f32fb9e20fe113573other"
    >
      <p
        class="MuiTypography-root MuiTypography-body2 css-rdc7yt-MuiTypography-root"
        title="0x320a50f32fb9e20fe113573other"
      >
        0x320a50f32fb9e20fe113573other
      </p>
    </a>
    <svg
      aria-hidden="true"
      aria-label="copy to clipboard"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-eibj8z-MuiSvgIcon-root"
      data-mui-internal-clone-element="true"
      data-testid="ContentCopyIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm3 4H8c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h11c1.1 0 2-.9 2-2V7c0-1.1-.9-2-2-2zm0 16H8V7h11v14z"
      />
    </svg>
  </div>
</div>
`;

exports[`render wallet link without link 1`] = `
<div>
  <div
    class="MuiBox-root css-1gvmo7"
  >
    <p
      class="MuiTypography-root MuiTypography-body2 css-rdc7yt-MuiTypography-root"
      title="0x320a50f32fb9e20fe113573unknown"
    >
      0x320a50f32fb9e20fe113573unknown
    </p>
    <svg
      aria-hidden="true"
      aria-label="copy to clipboard"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-eibj8z-MuiSvgIcon-root"
      data-mui-internal-clone-element="true"
      data-testid="ContentCopyIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm3 4H8c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h11c1.1 0 2-.9 2-2V7c0-1.1-.9-2-2-2zm0 16H8V7h11v14z"
      />
    </svg>
  </div>
</div>
`;

exports[`render wallet link without link when currentWallet passed 1`] = `
<div>
  <div
    class="MuiBox-root css-1gvmo7"
  >
    <p
      class="MuiTypography-root MuiTypography-body2 css-rdc7yt-MuiTypography-root"
      title="0x320a50f32fb9e20fe113573other"
    >
      0x320a50f32fb9e20fe113573other
    </p>
    <svg
      aria-hidden="true"
      aria-label="copy to clipboard"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-eibj8z-MuiSvgIcon-root"
      data-mui-internal-clone-element="true"
      data-testid="ContentCopyIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm3 4H8c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h11c1.1 0 2-.9 2-2V7c0-1.1-.9-2-2-2zm0 16H8V7h11v14z"
      />
    </svg>
  </div>
</div>
`;

exports[`render wallet with external link to scanner 1`] = `
<div>
  <div
    class="MuiBox-root css-1gvmo7"
  >
    <a
      class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1nrvlnd-MuiTypography-root-MuiLink-root"
      href="/wallet/0x320a50f32fb9e20fe113573"
      title="0x320a50f32fb9e20fe113573"
    >
      <p
        class="MuiTypography-root MuiTypography-body2 css-rdc7yt-MuiTypography-root"
        title="0x320a50f32fb9e20fe113573"
      >
        0x320a50f32fb9e20fe113573
      </p>
    </a>
    <svg
      aria-hidden="true"
      aria-label="copy to clipboard"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-eibj8z-MuiSvgIcon-root"
      data-mui-internal-clone-element="true"
      data-testid="ContentCopyIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm3 4H8c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h11c1.1 0 2-.9 2-2V7c0-1.1-.9-2-2-2zm0 16H8V7h11v14z"
      />
    </svg>
    <div
      class="MuiBox-root css-1isemmb"
    />
    <a
      class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1dkk3ib-MuiTypography-root-MuiLink-root"
      href="https://polygonscan.com/address/0x320a50f32fb9e20fe113573"
      rel="noopener noreferrer"
      target="_blank"
      title="https://polygonscan.com/address/0x320a50f32fb9e20fe113573"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-eibj8z-MuiSvgIcon-root"
        data-testid="OpenInNewIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
        />
      </svg>
    </a>
  </div>
</div>
`;

exports[`render wallet with short prop set: rerender with short prop 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -6,11 +6,11 @@
        class=\\"MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1nrvlnd-MuiTypography-root-MuiLink-root\\"
        href=\\"/wallet/0x320a50f32fb9e20fe113573\\"
        title=\\"0x320a50f32fb9e20fe113573\\"
      >
        <p
-         class=\\"MuiTypography-root MuiTypography-body2 css-rdc7yt-MuiTypography-root\\"
+         class=\\"MuiTypography-root MuiTypography-body2 css-18mpm0x-MuiTypography-root\\"
          title=\\"0x320a50f32fb9e20fe113573\\"
        >
          0x320a50f32fb9e20fe113573
        </p>
      </a>"
`;
